helix-term/src/commands.rs

```diff
@@ -4240,3 +4272,57 @@ fn suspend(_cx: &mut Context) {
     #[cfg(not(windows))]
     signal_hook::low_level::raise(signal_hook::consts::signal::SIGTSTP).unwrap();
 }
+
+// DAP
+fn dap_toggle_breakpoint(cx: &mut Context) {
+    use helix_lsp::block_on;
+
+    let (view, doc) = current!(cx.editor);
+    let text = doc.text().slice(..);
+    let pos = doc.selection(view.id).primary().cursor(text);
+
+    let breakpoint = helix_dap::SourceBreakpoint {
+        line: text.char_to_line(pos) + 1, // convert from 0-indexing to 1-indexing (TODO: could set debugger to 0-indexing on init)
```

Comment:
Not every debugger. Delve can't work with 0-based indexing.

Reply:
It can be specified via the initialize parameter: https://microsoft.github.io/debug-adapter-protocol/specification#Requests_Initialize

`linesStartAt1: false`

Reply:
No, Delve will error in this case as it has no support for 0-based

Reply:
Yeah I checked, okay, let's just support 1-based offsets then